modules/users/server/controllers/users.block.server.controller.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Module dependencies.
+ */
+const path = require('path');
+const errorService = require(path.resolve(
+  './modules/core/server/services/error.server.service',
+));
+const mongoose = require('mongoose');
+const User = mongoose.model('User');
+const log = require(path.resolve('./config/lib/logger'));
+
+/**
+ * Get the list of blocked users by the logged in user
+ */
+exports.getBlockedUsers = async function (req, res) {
+  try {
+    const user = await User.findById(req.user._id)
+      // Avoid pulling in sensitive fields from Mongoose
+      .select('-password -salt')
+      .populate({
+        path: 'blocked',
+        select: 'username displayName',
+        model: 'User',
+      });
+    res.send(user.blocked);
+  } catch (err) {
+    log('error', err);
+    return res.status(500).send({
+      message: 'invalid error',
+    });
+  }
+};
+
+/**
+ * Add a new userId to the blocked list
+ */
+exports.blockUser = async function (req, res) {
+  /*
+   * req.profile was instantiated by a prev middleware
+   */
+  if (!req.profile || req.profile._id === req.user._id) {
```

Comment:
Looks like `req.profile._id === req.user._id` is never `true`. So, currently, you can block your own profile:  

https://stackoverflow.com/questions/11637353/comparing-mongoose-id-and-stringsblocking

Reply:
Good catch! `_id` is Mongo `ObjectId` which has `equal` method:

https://github.com/Trustroots/trustroots/blob/46dc40602812521a0d6320f190f307eff40af244/modules/messages/server/controllers/messages.server.controller.js#L215-L216

Similarly validation is possible:
https://github.com/Trustroots/trustroots/blob/46dc40602812521a0d6320f190f307eff40af244/modules/messages/server/controllers/messages.server.controller.js#L457-L458